core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/CamelInternalProcessor.java

```diff
@@ -813,10 +801,10 @@ public void after(Exchange exchange, UnitOfWork uow) throws Exception {
                 UnitOfWorkHelper.doneUow(uow, exchange);
             }
 
-            // after UoW is done lets pop the route context which must be done on every existing UoW
             if (route != null && existing != null) {
                 existing.popRoute();
             }
+            // after UoW is done lets pop the route context which must be done on every existing UoW
```

Comment:
this comment better belongs where it was before

Reply:
Oh, alright. I'll merge and then move it back in place!